### PR TITLE
Initial tab layout with icons

### PR DESCRIPTION
**Pull request by Firius0408**
Initial tab layout of the app frontend based off of the figma designs.

### CODE REVIEW

**Comment by Firius0408** [comment]
Forgot to transfer the required node dependencies for React Navigation from my navigation branch. Furthermore, added icons to the tab bar.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
 
-This is a new update to the file.
+## Instructions
+
+```bash
+cd frontend
+npm install
+npm install --global expo-cli
+expo start
+```
```
Comment (Firius0408) [inline README.md:12]: Instructions for setting up and installing the frontend using expo

Comment (Firius0408) [inline README.md:12]: Instructions for installing and bootstrapping the frontend using expo

File: frontend/App.js [hunk 2]
```diff
@@ -1,21 +1,53 @@
-import { StatusBar } from 'expo-status-bar';
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import * as React from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { MaterialCommunityIcons, Ionicons, FontAwesome5 } from '@expo/vector-icons';
+import { DiagnosticsTab, DataTab, HomeTab, AmbianceTab, Start } from './homeTabs';
+import SettingsTab from './settingsTab';
```
Comment (Firius0408) [inline frontend/App.js:7]: Import the various modules required for React Navigation and tab icons.

File: frontend/App.js [hunk 3]
```diff
@@ -1,21 +1,53 @@
...
+          default:
+            return;
+        }
+      }
+    })}>
+      <Tab.Screen name="Diagnostics" component={DiagnosticsTab} />
+      <Tab.Screen name="Data" component={DataTab} />
+      <Tab.Screen name="Home" component={HomeTab} />
+      <Tab.Screen name="Ambiance" component={AmbianceTab} />
+      <Tab.Screen name="Settings" component={SettingsTab} />
```
Comment (Firius0408) [inline frontend/App.js:33]: Implement the different tabs to be displayed on the tab bar.

File: frontend/App.js [hunk 4]
```diff
@@ -1,21 +1,53 @@
...
+          case "Data":
+            return <MaterialCommunityIcons name="waveform" size={size} color={color} />;
+          case "Home":
+            return <Ionicons name="ios-globe-outline" size={size} color={color} />;
+          case "Ambiance":
+            return <FontAwesome5 name="lightbulb" size={size} color={color} />;
+          case "Settings":
+            return <Ionicons name="settings-outline" size={size} color={color} />;
+          default:
+            return;
```
Comment (Firius0408) [inline frontend/App.js:25]: Here is where the tab icons are implemented. I did this in the Tab.Navigator inside the screenOptions, but could easily be implemented at an individual Tab.Screen level.

File: frontend/App.js [hunk 5]
```diff
@@ -1,21 +1,53 @@
...
+const Tab = createBottomTabNavigator();
+
+function App() {
+  return (
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName="Home">
+        <Stack.Screen name="Start" component={Start} />
+        <Stack.Screen name="Home" component={HomeTabs} options={{ headerShown: false }} />
+      </Stack.Navigator>
+    </NavigationContainer>
```
Comment (Firius0408) [inline frontend/App.js:49]: Nested the Tab Navigator inside the Stack Navigator as per [React Navigation - hiding tab bar in specific screens](https://reactnavigation.org/docs/hiding-tabbar-in-screens)

This way, the Start screen does not display the bottom tab bar when navigated to.

File: frontend/homeTabs.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { Button, View, Text } from 'react-native';
+import { styles } from './style';
+
```
Comment (Firius0408) [inline frontend/homeTabs.js:4]: Implemented the various function components of each tab in homeTabs.js

File: frontend/settingsTab.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+function SettingsTab() {
+    return (
+        <SettingsStack.Navigator initialRouteName="Settings">
+            <SettingsStack.Screen name="Settings" component={Settings} />
+            <SettingsStack.Screen name="Preferences" component={Preferences} />
+            <SettingsStack.Screen name="Profile" component={Profile} />
+            <SettingsStack.Screen name="Notifications" component={Notifications} />
+            <SettingsStack.Screen name="Time Zone" component={TimeZone} />
+        </SettingsStack.Navigator>
```
Comment (Firius0408) [inline frontend/settingsTab.js:72]: Implement a Stack Navigator for the Settings page.

File: frontend/settingsTab.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+                    color: darkMode ? '#fff' : '#000'
+                }]}>Dark Mode</Text>
+                <View style={styles.darkModeToggleView}>
+                    <Switch onChange={() => setDarkMode(!darkMode)} value={darkMode} />
+                </View>
+            </View>
+            <Button onPress={() => navigation.navigate('Notifications')} title='Notifications' />
+            <Button onPress={() => navigation.navigate('Time Zone')} title='Time Zone' />
+        </View>
+    );
```
Comment (Firius0408) [inline frontend/settingsTab.js]: A mini implementation of dark mode. However, this only works on the Settings page and is currently a bug as per [Issue #9 ](https://github.com/williamchong256/immersive-sleep/issues/9)

File: frontend/settingsTab.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+}
+
```
Comment (Firius0408) [inline frontend/settingsTab.js:28]: Boilerplate implementations of the different settings

File: frontend/style.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+import { StyleSheet } from 'react-native';
+
```
Comment (Firius0408) [inline frontend/style.js:2]: Moved the StyleSheet out of App.js into its own file

**Review by Firius0408 — COMMENTED** [review]
A few comments providing more detailed explanations for why I implemented things a certain way and what some of the things do.

File: frontend/homeTabs.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    );
+}
+
+export function AmbianceTab() {
+    return (
+        <View style={styles.container}>
+            <Text>Boilerplate for AmbianceTab</Text>
+        </View>
+    );
+}
```
Comment (Firius0408) [inline frontend/homeTabs.js:44]: Implement boilerplates for each tab here.

File: frontend/settingsTab.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+function TimeZone() {
+    return (
+        <View style={styles.container}>
+            <Text>Boilerplate for TimeZone</Text>
+        </View>
+    );
+}
```
Comment (Firius0408) [inline frontend/settingsTab.js]: Implement boilerplates for each settings option here.

File: frontend/App.js (hunk 4 above)
Comment (courtneylau) [reply frontend/App.js:25]: I got a warning when I ran it and only the icon for Ambiance showed up:
"Warning: Failed prop type: invalid prop `name` of value `x` supplied to `Icon`" 
where x is waveform, ios-globe-outline, and settings-outline

Comment (Firius0408) [reply frontend/App.js:25]: @courtneylau Did you run npm install to install the necessary modules for the icons?

See commit [53176d8](https://github.com/williamchong256/immersive-sleep/pull/8/commits/53176d871b7d92bc4880d73ef5f2d63048c04d7a)

**Pull request merged**